### ISSUE TITLE
[pre-wip] document the API.

### ISSUE THREAD

**Issue by Lucarda** [pure-data/pddp#55]
in https://github.com/pure-data/pure-data/issues/1549#issuecomment-1014008639 @spacechild1 said:

> Sure. This just shows again how important it is to document an API 

May be this can be included in the https://github.com/pure-data/pddp/issues/35 "still in pre-alpha draft" project.

**Comment by Spacechild1** [comment]
When I spoke of documenting the API, I rather thought of using Doxygen comments. Of course, with `doxygen` you can automatically generate HTML documentation and that could be hosted on puredata.info.

**Comment by umlaeute** [comment]
i agree that having a literate programming documentation (e.g. using Doxygen; but even unstructured comments would already help) would be much more useful than having an independent document.

**Comment by danomatika** [comment]
I've added (light) Doxyen commenting to the modules I've written or updated. A big help would be to add such to m_pd.h. So far I prefer a terser form to keep readability in the headers themselves.

**Comment by Spacechild1** [comment]
> My experience pushing for internal documentation in OpenFrameworks also leads to not adopting an overly-pendantic or full-fledged style as A. most people won't bother themselves to follow it and B. if it looks hard most people won't bother to documentation write at all.

True, but here we are only talking about `m_pd.h` at the moment. It wouldn't be an ongoing undertaking (new functions are added very rarely), but rather a large one-time effort. Internal documentation would also be great, but there I would already be happy about regular code comments :-)

I think every function in `m_pd.h` should at least have `\brief`. We might also use `\param` and `\return` unless the arguments or return values are self-explanatory.

We could actually have more detailled documentation and usage examples in a seperate document and then in `m_pd.h` add links to the relevant sections. Maybe there is even a way to link to sections in the "Externals How-To" with Doxygen? This would be ideal because then we wouldn't have to maintain two seperate documents.

**Comment by Lucarda** [comment]
> Maybe there is even a way to link to sections in the "Externals How-To" with Doxygen?

Shouldn't be hard with CI or "webhooks" into the #35 repo. 

some auto CI or webhooks like:

git clone pd;
run doxygen to build on output-folder "foo"
clone #35
copy "foo" to #35/static/foo
run hugo for #35

The site is up including the doxygen part.

**Comment by danomatika** [comment]
> to start the schism early: i very much brefer `@brief` over `\brief` (i like LaTeX, but not in my code) 🤦

Ugg no. ;) As I've suggested, I'd prefer a minimal of *either*.
